### DISCUSSION TITLE
Installation with Manifests file Ready & Available shows Zero

### DISCUSSION THREAD

**Discussion by mohanss08** [nginx/kubernetes-ingress#3213] [Q&A] [answered]
I'm following this [Link][1] to install `nginx-ingress-controller` on my bare metal server with `Kubernetes-v1.24.7`

The below commands i have executed as part of installation.

```
$ git clone https://github.com/nginxinc/kubernetes-ingress.git --branch v2.4.1
$ cd kubernetes-ingress/deployments

$ kubectl apply -f common/ns-and-sa.yaml
$ kubectl apply -f rbac/rbac.yaml
$ kubectl apply -f rbac/ap-rbac.yaml
$ kubectl apply -f rbac/apdos-rbac.yaml

$ kubectl apply -f common/default-server-secret.yaml
$ kubectl apply -f common/nginx-config.yaml
$ kubectl apply -f common/ingress-class.yaml

$ kubectl apply -f daemon-set/nginx-ingress.yaml
```

I have followed `DaemonSet` method.

```
$ kubectl get all -n nginx-ingress
NAME                      READY   STATUS    RESTARTS   AGE
pod/nginx-ingress-2pdf9   0/1     Running   0          44s
pod/nginx-ingress-pc5pp   0/1     Running   0          44s
pod/nginx-ingress-pzhdf   0/1     Running   0          44s

NAME                           DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/nginx-ingress   3         3         0       3            0           <none>          44s
```

For all three `nginx-ingress` pods same error it shown.

```
$ kubectl describe pods nginx-ingress-2pdf9 -n nginx-ingress
Events:
  Type     Reason     Age                 From               Message
  ----     ------     ----                ----               -------
  Normal   Scheduled  92s                 default-scheduler  Successfully assigned nginx-ingress/nginx-ingress-2pdf9 to worker-3
  Normal   Pulled     91s                 kubelet            Container image "nginx/nginx-ingress:2.4.1" already present on machine
  Normal   Created    91s                 kubelet            Created container nginx-ingress
  Normal   Started    91s                 kubelet            Started container nginx-ingress
  Warning  Unhealthy  72s (x22 over 91s)  kubelet            Readiness probe failed: HTTP probe failed with statuscode: 503

```
```
$ kubectl logs -l app=nginx-ingress -n nginx-ingress
W1028 07:34:38.940508       1 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: failed to list *v1alpha1.TransportServer: the server could not find the requested resource (get transportservers.k8s.nginx.org)
E1028 07:34:38.940543       1 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: Failed to watch *v1alpha1.TransportServer: failed to list *v1alpha1.TransportServer: the server could not find the requested resource (get transportservers.k8s.nginx.org)
```
Still `READY` and `AVAILABLE` state showing `0`, Ideally it show `3` in both the categories. Please let me know what I'm missing here as part of installation?

I suspect problem could be due to `--branch v2.4.1` version, So please let me know the supported version for my Kubernetes.

Any help is appreciated.

  [1]: https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/

**Comment by haywoodsh** [comment, marked as answer]
Hi @mohanss08,

Would you mind sharing the `daemon-set/nginx-ingress.yaml` if you have modified it?

From your installation steps and error message, `the server could not find the requested resource (get transportservers.k8s.nginx.org)`. I am guessing that the Transport Server custom resource definition is missing from the cluster?

From the same page you linked to:

> Note: By default, it is required to create custom resource definitions for VirtualServer, VirtualServerRoute, TransportServer and Policy. Otherwise, the Ingress Controller pods will not become Ready. If you’d like to disable that requirement, configure [-enable-custom-resources](https://docs.nginx.com/nginx-ingress-controller/configuration/global-configuration/command-line-arguments#cmdoption-global-configuration) command-line argument to false and skip this section.

Please create the required CRDs by following [2. Create Custom Resources](https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/#create-custom-resources) in the documentation. Alternatively, if you are sure that you won't need those features provided by NGINX Ingress Controller, you can set the `-enable-custom-resources` command-line argument to `false` in your deployment YAML.

**Comment by mohanss08** [comment]
@haywoodsh - That was the issue, Thanks a lot for the quick help.

**Reply by jjngx** [reply]
You are welcome. We are closing the issue.
